blog/features/replica-signed-queries.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "Announcing replica signed queries"
+description: To enhance the trustworthiness of queries by providing better guarantees against malicious attacks, we're excited to announce the release of replica signed queries!
+tags: [New features]
+image: /img/blog/replica-signed-queries.jpg
+---
+
+When an update call is made to a canister on the Internet Computer, the response to includes a subnet signature that proves the data sent within the request is trustworthy. Query calls, however, are handled by single replicas and don't bear any information that validates their response is trustworthy, since their data could be tampered with by a malicious replica or malicious boundary node. As a workaround, certified data and certified variables can be used, which are embedded in responses to query calls that provide a layer of validation. However, using certified data and variables requires extra work for developers and sometimes does not fit into a developer's workflow seamlessly. 
+
+To enhance the trustworthiness of queries by providing better guarantees against malicious attacks, we're excited to announce the release of replica signed queries! 
+
+## What are replica signed queries?
+
+Replica signed queries are a new feature that introduces replica signatures in query responses. Every response to query call will contain a signature that signs specific parts of the response using the replica's private key. Then, query responses can be validated by fetching the replica's public key from the subnet's state tree, reading the public key of the signing replica, then verifying the signature.
+
+This new feature includes two primary changes: first, the subnet state tree now exposes the public keys of the replicas, then query responses will now contain a signature. 
+
+Having a single replica signature in a query response doesn't completely solve the problem that query responses cannot be trusted, however, replica signed queries do help improve the security of queries since the recipient can validate that the response came from a replica on the correct subnet. The primary goal of this feature is to provide better guarantees against malicious boundary nodes, since replica signed queries make it impossible for a malicious boundary node to return false data. 
+
+## How do replica signed queries work?
+
+First, a list of node IDs and their corresponding public keys is put into the subnet's state tree so that they can be certified by the subnet.
+
+Users can obtain the certificate to validate the node's keys through an HTTP `read_state` call that includes a timestamp value. 
+
+To support replica signed queries, the query response format has been changed. Previously, query responses used the following format:
+
+```
+pub enum HttpQueryResponse {
+    Replied {
+        reply: HttpQueryResponseReply,
+    },
+    Rejected {
+        error_code: String,
+        reject_code: u64,
+        reject_message: String,
+    },
+}
+```
```

Comment:
```suggestion
To support replica signed queries, the query response format has been changed. The response to a query call adds a list with one signature for the returned response produced by the IC node that evaluated the query call. 
```
Because the new response format simply adds one more field, we don't need to show it in detail twice. The new format is elaborated below.